src/components/drawer/ReusableDrawer.tsx

```diff
@@ -0,0 +1,50 @@
+import { Box, IconButton, Typography } from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import Drawer from '@mui/material/Drawer';
+import { FC } from 'react';
+
+interface ReusableDrawerProps {
+    open: boolean;
+    toggleDrawer: (open: boolean) => void;
+    children: JSX.Element
+    width?: number;
+    title: string
+}
+
+export const ReusableDrawer: FC<ReusableDrawerProps> = ({ open, toggleDrawer, children, width = 250, title }) => {
+
+    return (
+        <Drawer
+            sx={{ width: width }}
+            anchor='right'
+            open={open}
+            onClose={() => toggleDrawer(false)}
+        >
+            <Box
+                sx={{
+                    width: width,
+                    display: 'flex',
+                    justifyContent: 'space-between',
```

Comment:
FYI: zamiast space-between mozesz dac takie property jak `gap`. Okreslasz wtedy w pixelach odstep miedzy komponentami, `gap: 10px`. Czasem moze byc to lepsze. Nie mowie, ze tu, bo nie wiem jak wyglada, ale tylko na przyszlosc mowie :D